### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by Programazing**
Updated old methods using Linq.

### CODE REVIEW

File: bob/Bob.cs [hunk 1]
```diff
@@ -18,86 +18,76 @@ public static string Hey(string statement)
         response.Add(2, "Sure.");
         response.Add(3, "Fine. Be that way!");
 
+        if (ContainsNewLine(statement) == true)
+            return response[0].ToString();
```
Comment (togakangaroo) [inline bob/Bob.cs]: Why is `ToString()` necessary. The response is already a string.

File: bob/Bob.cs [hunk 2]
```diff
@@ -18,86 +18,76 @@ public static string Hey(string statement)
...
+
+        if (IsQuestionMarkNotAtEndOfSentence(statement) == true)
```
Comment (togakangaroo) [inline bob/Bob.cs]: No need to `== true` the method call already returns a `bool`

File: bob/Bob.cs [hunk 3]
```diff
@@ -18,86 +18,76 @@ public static string Hey(string statement)
...
         if (IsLowerCaseQuestion(statement) == true)
             return response[2].ToString();
 
-        //if (HasDoubleWhiteSpaceAfterWord(statement) == true)
-        //    return response[2].ToString();
-
-        //if (IsStringEmpty(statement) == true)
-        //    return response[3].ToString();
+        if (ContainsThreeOrMoreSentences(statement) == true)
+            return response[2].ToString();
```
Comment (togakangaroo) [inline bob/Bob.cs]: This is starting to get confusing with the numbers. What exactly is the point of the dictionary?

Couldn't you just put the responses in properly named variables and return those variables? This seems like a bunch of overhead for virtually no gain.

File: bob/Bob.cs [hunk 4]
```diff
@@ -18,86 +18,76 @@ public static string Hey(string statement)
...
 
-        return false;
-    }
-
-    private static bool HasDoubleWhiteSpaceAfterWord(string input)
+    private static bool ContainsThreeOrMoreSentences(string input)
     {
-        var pattern = new Regex(@"\w\s\s");
-        var match = pattern.Match(input);
+        string[] sentences = Regex.Split(input, @"[\.\!\?]");
```
Comment (togakangaroo) [inline bob/Bob.cs]: Just fyi, there's a `Split(char[])` method on `String` so you should be able to do `input.Split(new [] { '.', '!', '?'})`. Not a big deal but worth reaizing

File: bob/Bob.cs [hunk 5]
```diff
@@ -18,86 +18,76 @@ public static string Hey(string statement)
...
+        int count = 0;
+
+        foreach (string sentence in sentences)
+        {
+            count++;
+        }
 
-        if (match.Success)
+        if (count >= 3)
```
Comment (togakangaroo) [inline bob/Bob.cs]: Remember we talked about this. Any time you see this pattern of returning `true/false` based on the outcome of an `if()`, it is redundant

File: bob/Bob.cs [hunk 6]
```diff
@@ -18,86 +18,76 @@ public static string Hey(string statement)
...
-
-    private static bool HasDoubleWhiteSpaceAfterWord(string input)
+    private static bool ContainsThreeOrMoreSentences(string input)
     {
-        var pattern = new Regex(@"\w\s\s");
-        var match = pattern.Match(input);
+        string[] sentences = Regex.Split(input, @"[\.\!\?]");
+        int count = 0;
+
+        foreach (string sentence in sentences)
```
Comment (togakangaroo) [inline bob/Bob.cs]: `sentences.Count()`

**Review by togakangaroo — CHANGES_REQUESTED** [review]

File: bob/Bob.cs (hunk 3 above)
Comment (Programazing) [reply bob/Bob.cs]: You know you're right. I did that to try and easily keep track of the predefined responses and keep them all in one place.

File: bob/Bob.cs (hunk 2 above)
Comment (Programazing) [reply bob/Bob.cs]: I honestly forgot about that.

File: bob/Bob.cs (hunk 1 above)
Comment (Programazing) [reply bob/Bob.cs]: Just a straight up mistake on my part.

File: bob/Bob.cs (hunk 4 above)
Comment (Programazing) [reply bob/Bob.cs]: Very good to know.

File: bob/Bob.cs (hunk 6 above)
Comment (Programazing) [reply bob/Bob.cs]: I spaced out and forgot I was dealing with an array which has the Count() method.

File: bob/Bob.cs (hunk 5 above)
Comment (Programazing) [reply bob/Bob.cs]: I forgot I could enforce that in the return statement. Thank you for helping me with that by the way.

File: bob/Bob.cs [hunk 7]
```diff
@@ -7,97 +7,70 @@ public static class Bob
...
 
-    static bool ContainsNumberAndQuestionMark(string input)
+    private static bool ContainsThreeOrMoreSentences(string input)
     {
-        var pattern = new Regex(@"\d\?");
-        var match = pattern.Match(input);
-
-        if (match.Success)
-            return true;
+        string[] sentences = input.Split(new[] { '.', '!', '?' });
```
Comment (togakangaroo) [inline bob/Bob.cs]: this could also easily be a one liner

Comment (Programazing) [reply bob/Bob.cs]: You're right and I updated it.

**Review by togakangaroo — APPROVED** [review]
looks good

**Pull request merged**